datastore/indexmanager.go

```diff
@@ -87,6 +91,99 @@ func (im *IndexManager) run() {
 	}
 }
 
+// initMapping initializes ElasticSearch for cassabon.
+func (im *IndexManager) initMapping() {
+	mapping := map[string]map[string]map[string]map[string]map[string]string{
+		"mappings": map[string]map[string]map[string]map[string]string{
+			"path": map[string]map[string]map[string]string{
+				"properties": map[string]map[string]string{
+					"path": map[string]string{
+						"type":  "string",
+						"index": "not_analyzed",
+					},
+					"depth": map[string]string{
+						"type": "long",
+					},
+					"tenant": map[string]string{
+						"type": "string",
+					},
+					"leaf": map[string]string{
+						"type": "bool",
+					},
+				},
+			},
+		},
+	}
+
+	jsonMap, _ := json.Marshal(mapping)
+	config.G.Log.System.LogDebug("%s", string(jsonMap))
+
+	putreq, _ := http.NewRequest("PUT", config.G.ElasticSearch.PutURL, bytes.NewBuffer(jsonMap))
+	r := im.httpRequest(putreq)
+
+	if r == nil {
+		config.G.Log.System.LogFatal("Could not initialize mapping for ElasticSearch.")
+	}
+}
+
+// getAllLeafNodes queries ElasticSearch for all leaf nodes. Used for populating metric manager's stat paths on reboot.
+func (im *IndexManager) getAllLeafNodes() []string {
+	sort := []map[string]map[string]string{
+		{
+			"path": map[string]string{
+				"order": "asc",
+			},
+		},
+	}
+	query := map[string]map[string][]map[string]map[string]interface{}{
+		"bool": map[string][]map[string]map[string]interface{}{
+			"must": []map[string]map[string]interface{}{
+				{
+					"match": map[string]interface{}{
+						"leaf": true,
+					},
+				},
+			},
+		},
+	}
+
+	fullQuery := ERQuery{sort, query}
+	getreq := im.prepRequest(fullQuery)
+	r := im.httpRequest(getreq)
+
+	var esResp ElasticResponse
+	var pathList []string
+
+	if r != nil {
+		_ = json.Unmarshal(r, &esResp)
+
+		config.G.Log.System.LogDebug("esResp: %v", esResp)
+
+		for _, hit := range esResp.Hits.Hits {
+			pathList = append(pathList, hit.Source.Path)
+		}
+	} else {
+		logging.Statsd.Client.Inc("indexmgr.es.err.get", 1, 1.0)
+		config.G.Log.System.LogError("Error querying ES.")
+	}
+
+	return pathList
```

Comment:
You might want to issue a debug log entry indicating the number of paths retrieved, so you can see if that matches expectations.